SmartDeviceLink-ExampleSwift/ProxyManager.swift

```diff
@@ -0,0 +1,305 @@
+
+        // Start watching for a connection with a SDL Core
+        self.sdlManager?.start(readyHandler: { [unowned self] (success, error) in
+            if success {
+                delegate?.didChangeProxyState(ProxyState.connected)
+                self.addRPCObservers()
+                self.addPermissionManagerObservers()
+                print("SDL start file manager storage: \(self.sdlManager!.fileManager.bytesAvailable / 1024 / 1024) mb")
+            }
+            if let error = error {
+                print("Error starting SDL: \(error)")
+            }
+        })
+    }
+
+    private func setLifecycleConfigurationPropertiesOnConfiguration(_ configuration: SDLLifecycleConfiguration) -> SDLLifecycleConfiguration {
+        configuration.shortAppName = AppConstants.sdlShortAppName
+        configuration.appType = SDLAppHMIType.media()
+        configuration.appIcon = SDLArtwork.persistentArtwork(with: appIcon!, name: AppConstants.appIconName, as: .PNG)
+        return configuration
+    }
+
+    func send(request: SDLRPCRequest, responseHandler: SDLResponseHandler? = nil) {
+        guard sdlManager.hmiLevel != .none() else {
+            return
+        }
+        sdlManager.send(request, withResponseHandler: responseHandler)
+    }
+
+    func reset() {
+        sdlManager?.stop()
+        delegate?.didChangeProxyState(ProxyState.stopped)
+    }
+}
+
+// MARK: SDLManagerDelegate
+extension ProxyManager: SDLManagerDelegate {
+    func managerDidDisconnect() {
+        delegate?.didChangeProxyState(ProxyState.stopped)
+    }
+
+    func hmiLevel(_ oldLevel: SDLHMILevel, didChangeTo newLevel: SDLHMILevel) {
+        // On our first HMI level that isn't none, do some setup
+        if newLevel != .none() && firstHMIFull == true {
+            firstHMIFull = false
+        }
+        // HMI state is changing from NONE or BACKGROUND to FULL or LIMITED
+        if (oldLevel == .none() || oldLevel == .background()) && (newLevel == .full() || newLevel == .limited()) {
+            prepareRemoteSystem(overwrite: true) { [unowned self] in
+                self.showMainImage()
+                self.prepareButtons()
+                self.addSpeakMenuCommand()
+                self.addperformInteractionMenuCommand()
+                self.setText()
+                self.setDisplayLayout()
+                self.subscribeVehicleData()
+            }
+        }
+    }
+}
+
+// MARK: - Prepare Remote System
+extension ProxyManager {
+    fileprivate func addRPCObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(didReceiveVehicleData(_:)), name: .SDLDidReceiveVehicleData, object: nil)
+    }
+
+    fileprivate func addPermissionManagerObservers() {
+        _ = sdlManager.permissionManager.addObserver(forRPCs: ["SubscribeVehicleData"], groupType: .allAllowed) { (_, _) in
+        }
+    }
+
+    fileprivate func prepareRemoteSystem(overwrite: Bool = false, completionHandler: @escaping (Void) -> (Void)) {
+
```

Comment:
There seems to be some empty lines at the top of a few functions and classes which shouldn't be there.